calico/felix/actor.py

```diff
@@ -232,11 +250,31 @@ def _step(self):
                 except BaseException as e:
                     _log.exception("Exception processing %s", msg)
                     results.append(ResultOrExc(None, e))
+                    message_exception = e.__class__.__name__ +\
+                                                      ': ' + str(e)
                     _stats.increment("Messages executed with exception")
                 else:
+                    message_exception = None
                     results.append(ResultOrExc(result, None))
                     _stats.increment("Messages executed OK")
                 finally:
+                    batch_uuids.add(msg.uuid)
+                    # Log the message info
+                    # Strings are enclosed in quotes. Numbers aren't.
+                    _message_log.info("'%s': {"
+                                      "'received-time': %s, "
+                                      "'sender': '%s', "
+                                      "'receiver': '%s', "
+                                      "'function': '%s', "
+                                      "'tries': %s, "
+                                      "'exception': '%s'"
+                                      "}" % (msg.uuid,
+                                             int(time.time() * 1000),
+                                             msg.caller,
+                                             msg.recipient,
+                                             msg.name,
+                                             msg_tries[msg.uuid],
+                                             message_exception))
```

Comment:
This is fine, but perhaps it could be rewritten to use json.dumps() instead? My thinking:

* Using the json module is probably a bit more robust when it comes to quoting strings (for example, if there's an apostrophe in the message_exception variable)
* The keys and values would be directly together, rather than split via %s-formatting tokens. Reduces the chance that pairs are inadvertently swapped.
* It would be a little easier to add new fields here.

You can't quite do everything with the json module (`"key1": {...}` isn't valid JSON), but you can construct the big dictionary this way.